### ISSUE TITLE
ansible azure requirements are self-conflicting

### ISSUE THREAD

**Issue by ssbarnea** [ansible/ansible#53761]
##### SUMMARY
``packaging/requirements/requirements-azure.txt`` cannot even be installed alone without causing conflicts between its own list of packages.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
packaging

##### ANSIBLE VERSION
master, also appling to older versions and breaking the use of azure extra: `pip install ansible[azure]`

##### STEPS TO REPRODUCE

```bash
pip install packaging/requirements/requirements-azure.txt
```
##### EXPECTED RESULTS
``pip check`` should never report conflicts after installing the requriments, this check **must be part of the CI testing** to avoid ever publishing conflicting requrements.

##### ACTUAL RESULTS

```
pip check
azure-mgmt-web 0.32.0 has requirement msrestazure~=0.4.7, but you have msrestazure 0.5.0.
azure-mgmt-keyvault 0.40.0 has requirement msrestazure~=0.4.7, but you have msrestazure 0.5.0.
azure-mgmt-batch 4.1.0 has requirement msrestazure~=0.4.7, but you have msrestazure 0.5.0.
```

**Comment by ssbarnea** [comment]
The list from https://github.com/ansible/ansible/blob/devel/packaging/requirements/requirements-azure.txt does not make much sense to me as they are all pinned and in conflict.

**Comment by samdoran** [comment]
cc @abadger @nitzmahone

**Comment by Fred-sun** [comment]
fixes by #54443
